features/sharding/core/src/main/java/org/apache/shardingsphere/sharding/rule/checker/ShardingRuleChecker.java

```diff
@@ -64,7 +64,7 @@ public void check(final ShardingRuleConfiguration ruleConfig) {
     
     private void checkUniqueActualDataNodesInTableRules() {
         Collection<DataNode> uniqueActualDataNodes = new HashSet<>(shardingRule.getShardingTables().size(), 1F);
-        shardingRule.getShardingTables().forEach((key, value) -> checkUniqueActualDataNodes(uniqueActualDataNodes, key, value.getActualDataNodes().iterator().next()));
+        shardingRule.getShardingTables().forEach((key, value) -> value.getActualDataNodes().forEach(each -> checkUniqueActualDataNodes(uniqueActualDataNodes, key, each)));
```

Comment:
If `actualDataNodes` contains many nodes, checking all nodes will take more time. Is it possible to add a props to control whether to check all `actualDataNodes` or `sampleDataNode`?